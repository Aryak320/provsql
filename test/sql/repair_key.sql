\set ECHO none
\pset format csv
SET search_path TO provsql_test,provsql;

CREATE TABLE weather_conditions (dummy VARCHAR, weather VARCHAR,
                ground VARCHAR, p FLOAT);
INSERT INTO weather_conditions VALUES ('dummy',    'rain', 'wet', 0.35);
INSERT INTO weather_conditions VALUES ('dummy',    'rain', 'dry', 0.05);
INSERT INTO weather_conditions VALUES ('dummy', 'no rain', 'wet', 0.1);
INSERT INTO weather_conditions VALUES ('dummy', 'no rain', 'dry', 0.5);

SELECT repair_key('weather_conditions','dummy');
DO $$ BEGIN
  PERFORM set_prob(provenance(), p) FROM weather_conditions;
END $$;

CREATE TABLE result_repair_key AS
  SELECT *, probability_evaluate(provenance()) prob FROM (
    SELECT ground FROM weather_conditions GROUP BY ground) t;

SELECT remove_provenance('result_repair_key');
SELECT ground, ROUND(prob::numeric, 3) FROM result_repair_key;

DROP TABLE result_repair_key;
DROP TABLE weather_conditions;
