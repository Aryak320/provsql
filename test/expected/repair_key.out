\set ECHO none
 repair_key 
------------
 
(1 row)

 remove_provenance 
-------------------
 
(1 row)

 ground | round 
--------+-------
 dry    | 0.550
 wet    | 0.450
(2 rows)

